core/services/ocr2/plugins/directrequestocr/reporting.go

```diff
@@ -187,10 +187,20 @@ func (r *directRequestReporting) Report(ctx context.Context, ts types.ReportTime
 				commontypes.LogFields{"err": err, "observer": ob.Observer})
 			continue
 		}
+		seenReqIds := make(map[string]bool)
```

Comment:
A typical pattern for a Set is `map[string]struct{}`, because you never use bool value.
To set an item: `seenReqIds[id] = struct{}{}`.